### DISCUSSION TITLE
What will be the direction for supporting skinning for everything realted to overlays?

### DISCUSSION THREAD

**Discussion by ItsShamed** [ppy/osu#23522] [Q&A]
Looking at how #17616 and #21009 are making entire overlays skinnable/serialisable doesn't feel right to me. This means that basically components that could have the size of the entire playfield can be manually added by the end user, this doesn't make sense to me. 
But at the same time, there is currently no way (yet) to add skinning support for the overlays such as the `BreakOverlay` which imo should be housing #21009 and other things like stable's pass/fail overlay etc. 
How will that be handled in the future? Will there be a system similar to the "working layers" used in `Player`?
I am curious to hear thoughts about this because I feel like we will be hitting a wall sooner or <sub>*(more likely)*</sub> later because of too many overlays being skinnable...

> **Note**
> Asking this specifically for the dev practical sides of things

> **Note**
> I am aware that this is super-duper-mega low prio at the time I am writing this. But this is something that I think is to be considered at one point in time (no rushes though).

**Comment by peppy** [comment]
Yes, skinning targets are already supported, and have been implemented API wise (just not exposed from the skin editor pending usability concerns).

For instance, the break overlay will be its own skin target, and you will be able to customise what displays within it.

**Reply by ItsShamed** [reply]
Okay, thank you! Now I know that I am not the only who is thinking about making the break overlay a target layer. 
Now, I am still not convinced of making entire overlays being skinnable, is this the only path going forward or will there be any implementation alternatives that have not yet been and could be exploited?
